src/ScheduleFunctions.cpp

```diff
@@ -40,9 +54,12 @@ struct Container {
     int dim_idx;
     string name;
     Expr value;
+
+    Container(Type type, int dim_idx, string name, Expr value) :
+            type(type), dim_idx(dim_idx), name(std::move(name)), value(std::move(value)) {}
```

Comment:
Here and elsewhere, if these members are not mutated and the class doesn't need to be default-constructible better to hold them as const ref?

Reply:
It might well be, yes. I added the constructor so that I could use `emplace_back` instead of `push_back`. I kept copies where there were copies.

Reply:
No, these have to be copies. Temporaries are constructed and then passed in throughout `build_provide_loop_nest_helper`.